In [153]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [154]:
df = pd.read_csv("insurance.csv")
df[['sex','smoker','region']] = df[['sex','smoker','region']].astype('category') # coverted categegory column into category data type
df.head(3)

age     sex    bmi  children smoker     region     charges
0   19  female  27.90         0    yes  southwest  16884.9240
1   18    male  33.77         1     no  southeast   1725.5523
2   28    male  33.00         3     no  southeast   4449.4620

In [155]:
df.info()  # there is no null value

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype   
---  ------    --------------  -----   
 0   age       1338 non-null   int64   
 1   sex       1338 non-null   category
 2   bmi       1338 non-null   float64 
 3   children  1338 non-null   int64   
 4   smoker    1338 non-null   category
 5   region    1338 non-null   category
 6   charges   1338 non-null   float64 
dtypes: category(3), float64(2), int64(2)
memory usage: 46.3 KB


In [156]:
df["sex"].unique() # sex have two category i.e; male and female

['female', 'male']
Categories (2, object): ['female', 'male']

In [157]:
df["smoker"].unique() # smoker is categories into yes and no.

['yes', 'no']
Categories (2, object): ['no', 'yes']

In [158]:
df["region"].unique()  # there are four region i.e; 'southwest', 'southeast', 'northwest', 'northeast'

['southwest', 'southeast', 'northwest', 'northeast']
Categories (4, object): ['northeast', 'northwest', 'southeast', 'southwest']

In [159]:
df.columns  # there are 7 column ie; 'age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'.

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

In [160]:
df = df[['region','sex','smoker','age', 'bmi', 'children','charges']]
df.head(3)

region     sex smoker  age    bmi  children     charges
0  southwest  female    yes   19  27.90         0  16884.9240
1  southeast    male     no   18  33.77         1   1725.5523
2  southeast    male     no   28  33.00         3   4449.4620

In [161]:
df["region2"] = df['region'].str.slice(start=0, stop = 5) # created new column region2 with value north and south

In [162]:
df.rename(columns = {"region":"sub-region","region2":"region"},inplace = True)
df = df[['region','sub-region','sex','smoker','age', 'bmi', 'children','charges']]
df.head(3)

region sub-region     sex smoker  age    bmi  children     charges
0  south  southwest  female    yes   19  27.90         0  16884.9240
1  south  southeast    male     no   18  33.77         1   1725.5523
2  south  southeast    male     no   28  33.00         3   4449.4620

In [163]:
df['region'].unique()

array(['south', 'north'], dtype=object)

In [164]:
df[(df['smoker']=='yes') & (df['sex'] == 'male')]['age'].mean()  # average age of male who smokes.

38.44654088050314

In [165]:
df[(df['smoker']=='yes') & (df['sex'] == 'female')]['age'].mean()  # average age of female who smokes

38.608695652173914

In [166]:
len(df[(df['smoker']=='yes') & (df['sex'] == 'male')])  # Number of male who smokes

159

In [167]:
len(df[(df['smoker']=='yes') & (df['sex'] == 'female')])  # Number of female who smokes

115

In [168]:
df[(df['smoker']=='yes') & (df['sex'] == 'male')]['charges'].mean()  # average charge of male who smokes

33042.00597528303

In [169]:
df[(df['smoker']=='yes') & (df['sex'] == 'male')]['charges'].mean()  # average charge of female who smokes

33042.00597528303

In [170]:
smokers = len(df[(df['smoker']=='yes')])/len(df)*100  # total 20.47 % people smokes
smokers   

20.47832585949178

In [171]:
male_smokers = len(df[(df['smoker']=='yes') & (df['sex']=='male')])/len(df)*100  # Out of total 11.83% male smokes  
male_smokers

11.883408071748878

In [172]:
female_smokers = len(df[(df['smoker']=='yes') & (df['sex']=='female')])/len(df)*100 # and 8.59% female smokes
female_smokers

8.594917787742899

# insights:
1. there are 7 column ie; 'age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'.
2. there is no null value.
3. sex have two category i.e; male and female.
4. smoker is categories into yes and no.
5. average age of male who smokes is 38.446.
6. average age of female who smokes is 38.60.
7. Number of male who smokes is 159.
8. Number of female who smokes is 115.
9. average charge of male and female who smokes is equal, who smokes is 33042.
10. total 20.47 % people smokes, out of which 11.83% male and 8.59% female.